addons/mail/static/src/models/composer.js

```diff
@@ -140,6 +143,32 @@ registerModel({
                 composerView.update({ hasToRestoreContent: true });
             }
         },
+        async performRPClinkPreview(url) {
+            const linkPreviews = await this.messaging.rpc({
+                route: '/mail/link_preview',
+                params: { url, channel_id: this.activeThread.id },
+            }, { shadow: true });
+            return linkPreviews;
+        },
+        /**
+         * Try to fetch link preview when urls are inserted inside the composer.
+         */
+        async refreshLinkPreview(pastedText) {
+            const urls = pastedText.match(urlRegexp) || false;
+            if (!urls) {
+                return;
+            }
+            const existingUrls = this.linkPreviews.map(att => att.url);
```

Comment:
remove of existing url should be done even if no url are found now?

Reply:
The refresh doesn't remove anything. It's up to the user to remove the link preview once it's created.